lib/sensors/accelerometer.hpp

```diff
@@ -50,21 +51,37 @@ constexpr std::uint8_t kExpectedDeviceIdValue = 0x44;
 
 class Accelerometer : public II2cMuxSensor<core::RawAccelerationData> {
  public:
-  Accelerometer(core::ILogger &logger, io::II2c &i2c, const std::uint8_t channel);
+  static std::optional<Accelerometer> create(core::ILogger &logger,
+                                             std::shared_ptr<io::II2c> i2c,
+                                             const std::uint8_t channel,
+                                             const std::uint8_t device_address
+                                             = kDefaultAccelerometerAddress);
```

Comment:
why set a default address here? wouldn't this be better stated every time when creating the accelerometer

Reply:
dont understand your point here, this is mostly in-case we are told that electronics want to use an address other than the default, in which case we can provide another address if needed. (though this should be an enum ideally, will change)

Reply:
i feel like it's better practice to make it clear when calling what the address is instead of implying it sometimes and explicitly stating it other times